### PR TITLE
Inline constant parameters/members, remove unnecessary throws

### LINKED ISSUE #830: Remove unnecessary throws

### PR DESCRIPTION
**Pull request by amcp**
Fixes #830 and #831

Signed-off-by: Alexander Patrikalakis <amcp@amazon.co.jp>

-----

Thank you for contributing to JanusGraph!

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [x] Is there an issue associated with this PR? Is it referenced in the commit message?
- [x] Does your PR body contain #xyz where xyz is the issue number you are trying to resolve?
- [x] Has your PR been rebased against the latest commit within the target branch (typically `master`)?
- [x] Is your initial contribution a single, squashed commit?

### For code changes:
- [ ] Have you written and/or updated unit tests to verify your changes?
- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?
- [ ] If applicable, have you updated the LICENSE.txt file, including the main LICENSE.txt file in the root of this repository?
- [ ] If applicable, have you updated the NOTICE.txt file, including the main NOTICE.txt file found in the root of this repository?

### For documentation related changes:
- [ ] Have you ensured that format looks appropriate for the output in which it is rendered?
- [ ] If this PR is a documentation-only change, have you added a `[skip ci]`
  tag to the first line of your commit message to avoid spending CPU cycles in
  Travis CI when no code, tests, or build configuration are modified?

### Note:
Please ensure that once the PR is submitted, you check Travis CI for build issues and submit an update to your PR as soon as possible.

### CODE REVIEW

File: janusgraph-test/src/test/java/org/janusgraph/diskstorage/locking/ConsistentKeyLockerTest.java [hunk 1]
```diff
@@ -1086,26 +1082,26 @@ private StaticBuffer recordExceptionLockWrite(long duration, TemporalUnit tu, St
         StaticBuffer k = eq(defaultLockKey);
         final List<Entry> adds = eq(Collections.singletonList(add));
         final List<StaticBuffer> deletions;
-        if (null != del) {
-            deletions = eq(Collections.singletonList(del));
+        if (null != null) {
```
Comment (robertdale) [inline janusgraph-test/src/test/java/org/janusgraph/diskstorage/locking/ConsistentKeyLockerTest.java]: What??

Comment (amcp) [reply janusgraph-test/src/test/java/org/janusgraph/diskstorage/locking/ConsistentKeyLockerTest.java]: fixed

File: janusgraph-test/src/test/java/org/janusgraph/graphdb/idmanagement/IDManagementTest.java [hunk 2]
```diff
@@ -48,11 +48,11 @@
             IDManager.VertexIDType.PartitionedVertex, IDManager.VertexIDType.UnmodifiableVertex};
 
     @Before
-    public void setUp() throws Exception {
+    public void setUp() {
```
[the remark this answers is not in the document]
Comment (amcp) [reply janusgraph-test/src/test/java/org/janusgraph/graphdb/idmanagement/IDManagementTest.java]: fixed

File: janusgraph-test/src/test/java/org/janusgraph/graphdb/idmanagement/IDManagementTest.java [hunk 3]
```diff
@@ -48,11 +48,11 @@
...
     }
 
     @After
-    public void tearDown() throws Exception {
+    public void tearDown() {
```
[the remark this answers is not in the document]
Comment (amcp) [reply janusgraph-test/src/test/java/org/janusgraph/graphdb/idmanagement/IDManagementTest.java]: fixed

File: janusgraph-test/src/test/java/org/janusgraph/util/datastructures/BitMapTest.java [hunk 4]
```diff
@@ -26,7 +26,7 @@
 public class BitMapTest {
 
     @Before
-    public void setUp() throws Exception {
+    public void setUp() {
```
[the remark this answers is not in the document]
Comment (amcp) [reply janusgraph-test/src/test/java/org/janusgraph/util/datastructures/BitMapTest.java]: fixed

File: janusgraph-test/src/test/java/org/janusgraph/util/datastructures/BitMapTest.java [hunk 5]
```diff
@@ -43,7 +43,7 @@ public void testBitMap() {
     }
 
     @After
-    public void tearDown() throws Exception {
+    public void tearDown() {
```
[the remark this answers is not in the document]
Comment (amcp) [reply janusgraph-test/src/test/java/org/janusgraph/util/datastructures/BitMapTest.java]: fixed

File: janusgraph-es/src/main/java/org/janusgraph/diskstorage/es/ElasticSearchSetup.java [hunk 6]
```diff
@@ -52,11 +51,10 @@ public Connection connect(Configuration config) throws IOException {
...
     };
 
-    static void applySettingsFromJanusGraphConf(Map<String,Object> settings,
-                                                     Configuration config,
-                                                     ConfigNamespace rootNS) {
+    static void applySettingsFromJanusGraphConf(Map<String, Object> settings,
+                                                Configuration config) {
         int keysLoaded = 0;
-        Map<String,Object> configSub = config.getSubset(rootNS);
+        Map<String,Object> configSub = config.getSubset(ElasticSearchIndex.ES_CREATE_EXTRAS_NS);
```
Comment (sjudeng) [inline janusgraph-es/src/main/java/org/janusgraph/diskstorage/es/ElasticSearchSetup.java]: `final`

File: janusgraph-core/src/main/java/org/janusgraph/diskstorage/util/StaticArrayEntryList.java [hunk 7]
```diff
@@ -444,9 +444,9 @@ public Boolean get(byte[] array, int offset, int limit) {
     }
 
     //Copy-pasted from above replacing byte->long
-    private static long[] ensureSpace(long[] data, int offset, int length) {
-        if (offset+length<=data.length) return data;
-        long[] newData = new long[Math.max(data.length*2,offset+length)];
+    private static long[] ensureSpace(long[] data, int offset) {
+        if (offset+1<=data.length) return data;
+        long[] newData = new long[Math.max(data.length*2,offset+1)];
```
Comment (sjudeng) [inline janusgraph-core/src/main/java/org/janusgraph/diskstorage/util/StaticArrayEntryList.java]: `final`

File: janusgraph-core/src/main/java/org/janusgraph/graphdb/database/log/TransactionLogHeader.java [hunk 8]
```diff
@@ -100,14 +100,14 @@ public StaticBuffer serializeUserLog(Serializer serializer, Entry sourceTxEntry,
 
     public StaticBuffer serializePrimary(Serializer serializer, LogTxStatus status) {
         Preconditions.checkArgument(status==LogTxStatus.PRIMARY_SUCCESS || status==LogTxStatus.COMPLETE_SUCCESS);
-        DataOutput out = serializeHeader(serializer, 30, status);
+        DataOutput out = serializeHeader(serializer, status);
```
Comment (sjudeng) [inline janusgraph-core/src/main/java/org/janusgraph/graphdb/database/log/TransactionLogHeader.java]: `final`

File: janusgraph-core/src/main/java/org/janusgraph/graphdb/database/log/TransactionLogHeader.java [hunk 9]
```diff
@@ -100,14 +100,14 @@ public StaticBuffer serializeUserLog(Serializer serializer, Entry sourceTxEntry,
...
         return out.getStaticBuffer();
     }
 
     public StaticBuffer serializeSecondary(Serializer serializer, LogTxStatus status,
                                            Map<String,Throwable> indexFailures, boolean userLogSuccess) {
         Preconditions.checkArgument(status==LogTxStatus.SECONDARY_SUCCESS || status==LogTxStatus.SECONDARY_FAILURE);
-        DataOutput out = serializeHeader(serializer,30,status);
+        DataOutput out = serializeHeader(serializer, status);
```
Comment (sjudeng) [inline janusgraph-core/src/main/java/org/janusgraph/graphdb/database/log/TransactionLogHeader.java]: `final`

File: janusgraph-core/src/main/java/org/janusgraph/diskstorage/util/MetricInstrumentedStore.java [hunk 10]
```diff
@@ -210,30 +210,31 @@ private void recordSliceMetrics(StoreTransaction txh, List<Entry> row) {
         }
     }
 
-    static <T> T runWithMetrics(String prefix, String storeName, String name, IOCallable<T> impl) throws IOException {
```
[the remark this answers is not in the document]
Comment (amcp) [reply janusgraph-core/src/main/java/org/janusgraph/diskstorage/util/MetricInstrumentedStore.java]: fixed

File: janusgraph-core/src/main/java/org/janusgraph/diskstorage/util/MetricInstrumentedStore.java [hunk 11]
```diff
@@ -210,30 +210,31 @@ private void recordSliceMetrics(StoreTransaction txh, List<Entry> row) {
...
+    static <T> void runWithMetrics(String prefix, IOCallable<T> impl) throws IOException {
 
         if (null == prefix) {
-            return impl.call();
+            impl.call();
+            return;
         }
 
-        Preconditions.checkNotNull(name);
+        Preconditions.checkNotNull(MetricInstrumentedIterator.M_CLOSE);
```
[the remark this answers is not in the document]
Comment (amcp) [reply janusgraph-core/src/main/java/org/janusgraph/diskstorage/util/MetricInstrumentedStore.java]: fixed

File: janusgraph-core/src/main/java/org/janusgraph/graphdb/database/log/TransactionLogHeader.java (hunk 8 above)
Comment (amcp) [reply janusgraph-core/src/main/java/org/janusgraph/graphdb/database/log/TransactionLogHeader.java]: fixed

File: janusgraph-core/src/main/java/org/janusgraph/diskstorage/util/StaticArrayEntryList.java (hunk 7 above)
Comment (amcp) [reply janusgraph-core/src/main/java/org/janusgraph/diskstorage/util/StaticArrayEntryList.java]: fixed

File: janusgraph-core/src/main/java/org/janusgraph/graphdb/database/log/TransactionLogHeader.java (hunk 9 above)
Comment (amcp) [reply janusgraph-core/src/main/java/org/janusgraph/graphdb/database/log/TransactionLogHeader.java]: fixed

File: janusgraph-es/src/main/java/org/janusgraph/diskstorage/es/ElasticSearchSetup.java (hunk 6 above)
Comment (amcp) [reply janusgraph-es/src/main/java/org/janusgraph/diskstorage/es/ElasticSearchSetup.java]: fixed

**Pull request merged**